examples/charm++/load_balancing/kNeighbor/kNeighbor.C

```diff
@@ -170,20 +222,35 @@ class Block: public CBase_Block {
     int *neighbors;
     double *recvTimes;
     double startTime;
+    int msg_var;
 
     int random;
     int curIterMsgSize;
     int internalStepCnt;
     int sum;
+    int compute_work;
+    int compute_var;
+
+    int l_factor;
+    int l_frequency;
+    int l_max_frequency;
 
     toNeighborMsg **iterMsg;
 
   public:
     Block() {
-      //srand(thisIndex.x+thisIndex.y);
+      srand(thisIndex);
       usesAtSync = true;
 
-      numNeighbors = 2*STRIDEK;
+      numNeighbors = 2*comm_neigh;
+      msg_var = comm_var;
```

Comment:
Are we changing these values somewhere per chare array element? If not then we can just refer directly to the readonlies

Reply:
This is was done with the idea that we could change these values in the next iterations of this code. As of now, these values are not being changed.

Reply:
Ok, can you add some comments in the code, probably where these variables are defined or initialized, explaining that these values can be changed in the future for different LB testing.

In general, I think our plan when separating out the examples directory from the tests and benchmarks directories was so that examples/ would be exemplary Charm++ programs demonstrating different features or common patterns, tests/ would be correctness-oriented programs and benchmarks/ would be performance-oriented tests.